File: back/src/main/java/com/tdevilleduc/urthehero/back/model/Person.java
package com.tdevilleduc.urthehero.back.model;

import com.fasterxml.jackson.annotation.JsonIgnore;
import lombok.*;

import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;

@Entity
@NoArgsConstructor
@RequiredArgsConstructor
@Getter
@Setter
@ToString
@EqualsAndHashCode
public class Person {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @NonNull
    private Integer id;
    @NonNull
    private String login;
    @NonNull
    private String displayName;
    @NonNull
    private String email;

    @JsonIgnore
    private String password;

}

File: back/src/main/java/com/tdevilleduc/urthehero/back/model/Person.kt
package com.tdevilleduc.urthehero.back.model

import com.fasterxml.jackson.annotation.JsonIgnore
import io.swagger.v3.oas.annotations.media.Schema
import javax.persistence.Entity
import javax.persistence.GeneratedValue
import javax.persistence.GenerationType
import javax.persistence.Id

@Entity
data class Person(
        @Id
        @GeneratedValue(strategy = GenerationType.IDENTITY)
        @Schema(description = "\${swagger.model.person.param.storyId}")
        val id: Int,
        @Schema(description = "\${swagger.model.person.param.login}")
        var login: String = "",
        @Schema(description = "\${swagger.model.person.param.displayName}")
        var displayName: String = "",
        @Schema(description = "\${swagger.model.person.param.email}")
        var email: String = "") {

    @JsonIgnore
    var password: String? = null

}